--- gdm-3.18.3/common/gdm-log.c.orig	2016-07-30 15:49:59.644284962 +0700
+++ gdm-3.18.3/common/gdm-log.c	2016-07-30 15:50:49.306528265 +0700
@@ -30,7 +30,7 @@
 #include <unistd.h>
 
 #include <syslog.h>
-#include <systemd/sd-daemon.h>
+// #include <elogind/sd-daemon.h>
 
 #include <glib.h>
 #include <glib/gstdio.h>
--- gdm-3.18.3/common/gdm-common.c.orig	2016-07-30 15:55:02.347766443 +0700
+++ gdm-3.18.3/common/gdm-common.c	2016-07-30 15:55:20.123853333 +0700
@@ -39,7 +39,7 @@
 #include "mkdtemp.h"
 #endif
 
-#include <systemd/sd-login.h>
+// #include <elogind/sd-login.h>
 
 #define GDM_DBUS_NAME                            "org.gnome.DisplayManager"
 #define GDM_DBUS_LOCAL_DISPLAY_FACTORY_PATH      "/org/gnome/DisplayManager/LocalDisplayFactory"
@@ -459,7 +459,7 @@
            If none are found, create a new one. */
 
         /* Note that we mostly use free () here, instead of g_free ()
-         * since the data allocated is from libsystemd-logind, which
+         * since the data allocated is from libelogind-logind, which
          * does not use GLib's g_malloc (). */
 
         res = sd_pid_get_session (0, &our_session);
--- gdm-3.18.3/daemon/gdm-server.c.orig	2016-07-30 15:51:11.306636018 +0700
+++ gdm-3.18.3/daemon/gdm-server.c	2016-07-30 15:52:29.020016484 +0700
@@ -43,10 +43,10 @@
 #include <linux/vt.h>
 #endif
 
-#include <systemd/sd-daemon.h>
+// #include <elogind/sd-daemon.h>
 
 #ifdef ENABLE_SYSTEMD_JOURNAL
-#include <systemd/sd-journal.h>
+#include <elogind/sd-journal.h>
 #endif
 
 #include <glib/gi18n.h>
--- gdm-3.18.3/daemon/gdm-session-worker.c.orig	2016-07-30 15:52:45.852098855 +0700
+++ gdm-3.18.3/daemon/gdm-session-worker.c	2016-07-30 15:53:07.260203607 +0700
@@ -49,10 +49,10 @@
 
 #include <X11/Xauth.h>
 
-#include <systemd/sd-daemon.h>
+// #include <elogind/sd-daemon.h>
 
 #ifdef ENABLE_SYSTEMD_JOURNAL
-#include <systemd/sd-journal.h>
+#include <elogind/sd-journal.h>
 #endif
 
 #ifdef HAVE_SELINUX
--- gdm-3.18.3/daemon/gdm-session-worker-job.c.orig	2016-07-30 15:53:28.284306464 +0700
+++ gdm-3.18.3/daemon/gdm-session-worker-job.c	2016-07-30 15:53:48.300404372 +0700
@@ -36,10 +36,10 @@
 #include <sys/prctl.h>
 #endif
 
-#include <systemd/sd-daemon.h>
+// #include <elogind/sd-daemon.h>
 
 #ifdef ENABLE_SYSTEMD_JOURNAL
-#include <systemd/sd-journal.h>
+#include <elogind/sd-journal.h>
 #endif
 
 #include <glib.h>
--- gdm-3.18.3/daemon/gdm-manager.c.orig	2016-07-30 15:54:11.514517906 +0700
+++ gdm-3.18.3/daemon/gdm-manager.c	2016-07-30 15:54:42.636670082 +0700
@@ -34,7 +34,7 @@
 #include <glib/gstdio.h>
 #include <glib-object.h>
 
-#include <systemd/sd-login.h>
+#include <elogind/sd-login.h>
 
 #include "gdm-common.h"
 
@@ -215,7 +215,7 @@
                 g_set_error (error,
                              GDM_DISPLAY_ERROR,
                              GDM_DISPLAY_ERROR_GETTING_SESSION_INFO,
-                             "Error getting session id from systemd: %s",
+                             "Error getting session id from elogind: %s",
                              g_strerror (-ret));
                 return NULL;
         }
@@ -242,7 +242,7 @@
                 g_set_error (error,
                              GDM_DISPLAY_ERROR,
                              GDM_DISPLAY_ERROR_GETTING_SESSION_INFO,
-                             "Error getting uid for session id %s from systemd: %s",
+                             "Error getting uid for session id %s from elogind: %s",
                              session_id,
                              g_strerror (-ret));
                 return FALSE;
@@ -277,7 +277,7 @@
                 g_set_error (error,
                              GDM_DISPLAY_ERROR,
                              GDM_DISPLAY_ERROR_GETTING_SESSION_INFO,
-                             "Error getting class for session id %s from systemd: %s",
+                             "Error getting class for session id %s from elogind: %s",
                              session_id,
                              g_strerror (-ret));
                 return FALSE;
@@ -431,7 +431,7 @@
                 g_set_error (error,
                              GDM_DISPLAY_ERROR,
                              GDM_DISPLAY_ERROR_GETTING_SESSION_INFO,
-                             "Error getting uid for session id %s from systemd: %s",
+                             "Error getting uid for session id %s from elogind: %s",
                              session_id,
                              g_strerror (-ret));
                 out_seat = NULL;
@@ -458,7 +458,7 @@
                 g_set_error (error,
                              GDM_DISPLAY_ERROR,
                              GDM_DISPLAY_ERROR_GETTING_SESSION_INFO,
-                             "Error getting tty for session id %s from systemd: %s",
+                             "Error getting tty for session id %s from elogind: %s",
                              session_id,
                              g_strerror (-ret));
                 out_tty = NULL;
@@ -1752,7 +1752,7 @@
 get_display_device (GdmManager *manager,
                     GdmDisplay *display)
 {
-        /* systemd finds the display device out on its own based on the display */
+        /* elogind finds the display device out on its own based on the display */
         return NULL;
 }
 
--- gdm-3.18.3/libgdm/gdm-user-switching.c.orig	2016-07-30 15:55:35.451928247 +0700
+++ gdm-3.18.3/libgdm/gdm-user-switching.c	2016-07-30 15:55:50.428001429 +0700
@@ -31,7 +31,7 @@
 #include <glib-object.h>
 #include <gio/gio.h>
 
-#include <systemd/sd-login.h>
+#include <elogind/sd-login.h>
 
 #include "common/gdm-common.h"
 #include "gdm-user-switching.h"
@@ -200,7 +200,7 @@
            If none are found, create a new one. */
 
         /* Note that we mostly use free () here, instead of g_free ()
-         * since the data allocated is from libsystemd-logind, which
+         * since the data allocated is from libelogind-logind, which
          * does not use GLib's g_malloc (). */
 
         res = sd_pid_get_session (0, &our_session);
