--- ./gnome-session/main.c.orig	2017-03-09 01:49:10.015919616 +0700
+++ ./gnome-session/main.c	2017-04-30 02:14:20.258703772 +0700
@@ -41,10 +41,6 @@
 #include "gsm-system.h"
 #include "gsm-fail-whale.h"
 
-#ifdef HAVE_SYSTEMD
-#include <systemd/sd-journal.h>
-#endif
-
 #define GSM_DBUS_NAME "org.gnome.SessionManager"
 
 static gboolean failsafe = FALSE;
@@ -318,22 +314,6 @@
 
         g_option_context_free (options);
 
-        /* Rebind stdout/stderr to the journal explicitly, so that
-         * journald picks ups the nicer "gnome-session" as the program
-         * name instead of whatever shell script GDM happened to use.
-         */
-#ifdef HAVE_SYSTEMD
-        if (!debug) {
-                int journalfd;
-
-                journalfd = sd_journal_stream_fd (PACKAGE, LOG_INFO, 0);
-                if (journalfd >= 0) {
-                        dup2(journalfd, 1);
-                        dup2(journalfd, 2);
-                }
-        }
-#endif
-
         gdm_log_init ();
         gdm_log_set_debug (debug);
 
