diff -Naur a/tools/dbusmenu-dumper.c b/tools/dbusmenu-dumper.c
--- a/tools/dbusmenu-dumper.c	2012-10-03 22:54:22.000000000 +0700
+++ b/tools/dbusmenu-dumper.c	2013-06-23 19:15:34.414109953 +0700
@@ -388,7 +388,9 @@
 int
 main (int argc, char ** argv)
 {
-	g_type_init();
+#if !GLIB_CHECK_VERSION(2, 36, 0)
+	g_type_init();  // Deprecated in glib 2.36.0
+#endif
 	GError * error = NULL;
 	GOptionContext * context;
 
diff -Naur a/tools/testapp/main.c b/tools/testapp/main.c
--- a/tools/testapp/main.c	2012-10-03 22:54:22.000000000 +0700
+++ b/tools/testapp/main.c	2013-06-23 19:16:29.670108522 +0700
@@ -135,7 +135,9 @@
 
 int main (int argc, char ** argv)
 {
-	g_type_init();
+#if !GLIB_CHECK_VERSION(2, 36, 0)
+	g_type_init();  // Deprecated in glib 2.36.0
+#endif
 
 	if (argc != 2) {
 		g_warning(USAGE);
